test/apiv2/26-containersWait.at

```diff
@@ -11,6 +11,46 @@ podman pull "${IMAGE}" &>/dev/null
 # Ensure clean slate
 podman rm -a -f &>/dev/null
 
+# Make sure that the Docker wait endpoint waits for the container to transition
+# into exited state (and not stopped).
+#
+# Reported by Gitlab: https://gitlab.com/gitlab-org/gitlab-runner/-/issues/27119#note_970712864
+#
+# FIXME: the test also passes with previous versions of Podman (i.e., v4.1.0).
+#        We need to improve the test to make sure we won't regress in the future.
+
+podman run --replace --name=gitlab -d $IMAGE top
+t GET libpod/containers/json 200 length=1
+t GET  containers/gitlab/json 200 \
+  .State.Running=true
+
+podman stop gitlab &
+
+t POST "containers/gitlab/wait?condition=not-running" 200
+t GET  containers/gitlab/json 200 \
+  .State.Status~\\\(exited\\\) \
+  .State.Running=false
```

Comment:
@edsantiago, I need your guidance.  I don't manage to create a reproducer.

What I attempt to do is to reproduce the gitlab test. We observed that the containers were in the `stopped` state, even when waiting for `exited` - the two were used interchangeably.

What I expect to see is that at least some runs of the tests with an older podman would fail, but they don't.

Reply:
@vrothberg I'm having trouble grokking the full context here (still catching up from the weekend). However, do you think the lines below have anything to do with your PR? And if so, could you disable those workarounds?

https://github.com/containers/podman/blob/9f1bd0a0a1494f46a49ca7f22511c5a78006afd8/test/system/050-stop.bats#L15-L20

https://github.com/containers/podman/blob/9f1bd0a0a1494f46a49ca7f22511c5a78006afd8/test/system/030-run.bats#L688-L693